### PR TITLE
Double check Ns in fail.vcf

### PR DESCRIPTION
**Pull request by DarianHole**
Double checks and corrects N's in fail vcf with python script and bcftools

Additional automation of uploading to IRIDA available

### CODE REVIEW

File: bin/correct_n.py [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+                        (true) or the end of the sequence (false)
+    
+    RETURNS:
+        `integer` position of the first non-N
+    '''
+
+    if start:
+        # Nucleotide location of first non-N is equal to number of N's at the start
+        for position, char in enumerate(sequence):
+            if char != 'N':
```
Comment (Takadonet) [inline bin/correct_n.py:126]: perhaps ensure that the char variable is upper case. Just in case one day we get 'n'

File: bin/correct_n.py [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+                return position
+    else:
+        # Iteration from the end of the sequence, position of first N is equal to the length
+        # of the sequence minus the first location that is not an N when iterating from the end
+        for position, char in enumerate(sequence[::-1]):
+            if char != 'N':
```
Comment (Takadonet) [inline bin/correct_n.py:132]: perhaps ensure that the char variable is upper case. Just in case one day we get 'n'

File: conf/nanopore.config [hunk 3]
```diff
@@ -18,6 +18,9 @@ params {
     // Output cram instead of bam files
     outCram = false
 
+    // Experimental - check if the Ns in the fail VCF file have enough evidence to support a reference basecall
+    correctN = false
```
Comment (Takadonet) [inline conf/nanopore.config]: change to default to true. It will be a new folder anyway so users can see both the original and the correted

**Review by Takadonet — CHANGES_REQUESTED** [review]

File: conf/nanopore.config (hunk 3 above)
Comment (DarianHole) [reply conf/nanopore.config]: done

File: bin/correct_n.py (hunk 1 above)
Comment (DarianHole) [reply bin/correct_n.py:126]: [line 46](https://github.com/phac-nml/ncov2019-artic-nf/pull/47/files#diff-1707c73f6b57cda34c4515cba8c57400R46) returns the uppercase string of the sequence so it should always be uppercase

**Review by Takadonet — APPROVED** [review]

**Review by Takadonet — APPROVED** [review]

**Pull request merged**